modules/diskq/diskq-grammar.ym

```diff
@@ -64,13 +64,21 @@ DiskQueueOptions *last_options;
 %token KW_QOUT_SIZE
 %token KW_DIR
 
+%type <num> nonnegative_integer64
 
 %%
 
 start
 	: LL_CONTEXT_INNER_DEST dest_diskq { YYACCEPT; }
 	;
 
+nonnegative_integer64
+  : LL_NUMBER
```

Comment:
If it is handled at the grammar level. (Is it possible to handle by the lexer ?! Or does that make sense ?!)
I would make the not specific version `nonnegative_integer` just an alias of the specific `nonnegative_integer64` or `nonnegative_integer32` (but maybe not the later one).

```latex
nonnegative_integer64
  : LL_NUMBER
    {
      CHECK_ERROR(($1 >= 0), @1, "It cannot be negative");
    }

nonnegative_integer
  : nonnegative_integer64
    {
      CHECK_ERROR(($1 <= G_MAXINT32), @1, ...);
    }
```

Reply:
No, it is not possible to handle it on the lexer level, but making `nonnegative_integer` a special case of `nonnegative_integer64` makes sense. Thanks!